api/controllers/v1/cave/find.js

```diff
@@ -1,55 +1,64 @@
-const CaverService = require('../../../services/CaverService');
 const CaveService = require('../../../services/CaveService');
 const ControllerService = require('../../../services/ControllerService');
 const DescriptionService = require('../../../services/DescriptionService');
+const DocumentService = require('../../../services/DocumentService');
 const NameService = require('../../../services/NameService');
 const { toCave } = require('../../../services/mapping/converters');
 
 module.exports = async (req, res) => {
-  TCave.findOne(req.params.id)
-    .populate('descriptions')
-    .populate('documents')
-    .populate('descriptions')
-    .populate('entrances')
-    .populate('histories')
-    .populate('id_author')
-    .populate('id_reviewer')
-    .populate('names')
-    .exec(async (err, found) => {
-      const params = {};
-      params.controllerMethod = 'CaveController.find';
-      params.searchedItem = `Cave of id ${req.params.id}`;
-      params.notFoundMessage = `${params.searchedItem} not found.`;
-      if (err) {
-        return res.serverError(
-          `An unexpected server error occured when trying to get ${params.searchedItem}`
-        );
-      }
-      let caveResult;
-      if (found) {
-        caveResult = found;
-        caveResult.massifs = await CaveService.getMassifs(caveResult.id);
-        await CaveService.setEntrances([caveResult]);
-        await NameService.setNames([caveResult], 'cave');
-        await NameService.setNames(caveResult?.entrances, 'entrance');
-        await NameService.setNames(caveResult?.massifs, 'massif');
-        caveResult.descriptions = await DescriptionService.getCaveDescriptions(
-          caveResult.id
-        );
-        caveResult.histories.map(async (h) => {
-          // eslint-disable-next-line no-param-reassign
-          h.author = await CaverService.getCaver(h.author, req);
-          return h;
-        });
-      }
+  const params = {
+    controllerMethod: 'CaveController.find',
+    searchedItem: `Cave of id ${req.params.id}`,
+    notFoundMessage: `Cave of id ${req.params.id} not found.`,
+  };
 
-      return ControllerService.treatAndConvert(
-        req,
-        err,
-        caveResult,
-        params,
-        res,
-        toCave
-      );
-    });
+  try {
+    const cave = await TCave.findOne(req.params.id)
+      .populate('descriptions')
+      .populate('documents')
+      .populate('entrances')
+      .populate('author')
+      .populate('reviewer')
+      .populate('names');
+
+    // TODO How to delete/restore entity ?
```

Comment:
I don't understand this comment: why we should be able to delete/restore in the cave/find controller?

Reply:
Like in the previous refactor on other entries it serves as a reminder that cave delete / restore is not currently implemented.

Reply:
Ok but why mentioning it here ? In the find code ? I don't get it 😅